src/error_handling.c

```diff
@@ -195,7 +195,7 @@ format_error_string(void)
   if (!__rcutils_error_is_set(__rcutils_error_state)) {
     return;
   }
-  size_t bytes_it_would_have_written = snprintf(
+  size_t bytes_it_would_have_written = (size_t)snprintf(
```

Comment:
There's actually a few problems here.  `snprintf` is defined to return an integer, because the return value can be negative on failure.  So the return value should be stored in an integer.  After that, `bytes_it_would_have_written` should be checked for negative; if it failed, we definitely don't want to continue on and pass bogus sizes around as unsigned.  Finally, we would need a cast from `int` to `size_t` while passing `bytes_it_would_have_written` into `allocator.allocate()` and the lower-down `snprintf` to avoid the warnings.